pkg/common/version/version.go

```diff
@@ -7,12 +7,17 @@ const (
 )
 
 var (
-	gittag = ""
+	gittag  = ""
+	githash = ""
 )
 
 func Version() string {
 	if gittag == "" {
-		return fmt.Sprintf("%s-dev", Base)
+		// Metrics systems sometimes automatically reject values with full SHA1 hashes
+		// since they appear to be high cardinality.
+		// For our purposes, the first 7 chars of the SHA should be enough.
+		githashShort := githash[:7]
```

Comment:
`gittag` and `githash` are only set when built via the Makefile. Slicing `githash` will panic when unset (i.e. during tests).